src/Http.hs

```diff
@@ -17,13 +23,31 @@ import           Network.HTTP.Types
 import           Network.Wai.Handler.Warp
 import           Network.Socket
 
-start :: String -> IO (Bool, String) -> IO ()
-start socketName trigger = do
+socketName :: String
+socketName = ".autospec.sock"
+
+socketAddr :: SockAddr
+socketAddr = SockAddrUnix socketName
+
+newSocket :: IO Socket
+newSocket = socket AF_UNIX Stream 0
+
+withSocket :: (Socket -> IO a) -> IO a
+withSocket action = bracket newSocket sClose action
+
+withServer :: IO (Bool, String) -> IO a -> IO a
```

Comment:
I find `withServer` hard to understand. I don't know if it could be done much better though, but maybe a comment would be nice explaining when `trigger` is executed and when `action` is.

Or: make `withServer :: Application -> IO a -> IO a`. And just use `app` from below everywhere you call `withServer`.